rust/connlib/tunnel/src/dns.rs

```diff
@@ -63,11 +63,10 @@ impl<T, V> ResolveStrategy<T, DnsQueryParams, V> {
 pub(crate) fn parse<'a>(
     dns_resources: &HashMap<String, ResourceDescriptionDns>,
     dns_resources_internal_ips: &HashMap<DnsResource, HashSet<IpAddr>>,
+    dns_mapping: &bimap::BiMap<IpAddr, DnsServer>,
```

Comment:
The bimap is between a public-routable IP of an upstream DNS server, and a sentinel in our CGNAT space, right?

Reply:
not neccesarily public, it could be a private, but yeah, it's between the real dns and our sentinel ips to capture traffic.